src/RelayBoardV3.cpp

```diff
@@ -284,65 +284,89 @@ void RelayBoardV3::handle(std::shared_ptr<const pilot::kinematics::omnidrive::Dr
-		out->effort[5] = value->steer_torque.get(pilot::kinematics::position_code_e::BACK_RIGHT);
-		out->effort[6] = value->drive_torque.get(pilot::kinematics::position_code_e::FRONT_RIGHT);
-		out->effort[7] = value->steer_torque.get(pilot::kinematics::position_code_e::FRONT_RIGHT);
+
+	// Containers for detected joints
+	std::vector<std::string> joint_names;
+	std::vector<double> joint_positions;
+	std::vector<double> joint_velocities;
+	std::vector<double> joint_efforts;
+
+	// Define all possible module positions
+	std::vector<pilot::kinematics::position_code_e> possible_positions = {
+		pilot::kinematics::position_code_e::FRONT_LEFT,
+		pilot::kinematics::position_code_e::BACK_LEFT,
+		pilot::kinematics::position_code_e::BACK_RIGHT,
+		pilot::kinematics::position_code_e::FRONT_RIGHT
+	};
+
+	// Detect available drive modules
+	for (const auto& pos : possible_positions) {
+		// Using steering positions to determine availability of a module
+		if (value->steer_pos.positions.count(pos)) {
+			// Adding the drive wheel velocity
+			switch (pos) {
+				case pilot::kinematics::position_code_e::FRONT_LEFT:
+					joint_names.push_back("wheel_front_left_joint");
+					break;
+				case pilot::kinematics::position_code_e::BACK_LEFT:
+					joint_names.push_back("wheel_back_left_joint");
+					break;
+				case pilot::kinematics::position_code_e::BACK_RIGHT:
+					joint_names.push_back("wheel_back_right_joint");
+					break;
+				case pilot::kinematics::position_code_e::FRONT_RIGHT:
+					joint_names.push_back("wheel_front_right_joint");
+					break;
+			}
+			joint_positions.push_back(value->drive_pos.positions.count(pos) ? value->drive_pos.get(pos) : 0.0);
+			joint_velocities.push_back(value->drive_vel.get(pos));
+
+			// Adding the steering caster position
+			switch (pos) {
+				case pilot::kinematics::position_code_e::FRONT_LEFT:
+					joint_names.push_back("caster_front_left_joint");
+					break;
+				case pilot::kinematics::position_code_e::BACK_LEFT:
+					joint_names.push_back("caster_back_left_joint");
+					break;
+				case pilot::kinematics::position_code_e::BACK_RIGHT:
+					joint_names.push_back("caster_back_right_joint");
+					break;
+				case pilot::kinematics::position_code_e::FRONT_RIGHT:
+					joint_names.push_back("caster_front_right_joint");
+				break;
+			}
+			joint_positions.push_back(value->steer_pos.get(pos));
+			joint_velocities.push_back(value->steer_vel.velocities.count(pos) ? value->steer_vel.get(pos) : 0.0);
+		}
+	}
+
+	// Resize the ROS message arrays based on detected joints
+	size_t num_joints = joint_names.size();
+	out->name.resize(num_joints);
+	out->position.resize(num_joints);
+	out->velocity.resize(num_joints);
+
+	if (value->has_torque) {
+		out->effort.resize(num_joints);
 	}
+
+	// Fill the ROS message with detected values
+	for (size_t i = 0; i < num_joints; i++) {
+		out->name[i] = joint_names[i];
+		out->position[i] = joint_positions[i];
+		out->velocity[i] = joint_velocities[i];
+
+		if (value->has_torque) {
```

Comment:
I would collect these above as `joint_efforts` and handle them the same way as you handled the `joint_positions`.